homeassistant/components/sensibo/__init__.py

```diff
@@ -28,3 +36,22 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
             del hass.data[DOMAIN]
         return True
     return False
+
+
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    # Change entry unique id from api_key to username
+    if entry.version == 1:
+        api_key = entry.data[CONF_API_KEY]
+        new_unique_id = await async_validate_api(hass, api_key)
+        if new_unique_id in [INVALID_AUTH, CANNOT_CONNECT, NO_DEVICES, NO_USERNAME]:
+            return False
+
+        LOGGER.debug("Migrate Sensibo config entry unique id to %s", new_unique_id)
+        hass.config_entries.async_update_entry(
+            entry,
+            unique_id=new_unique_id,
+        )
+        entry.version = 2
```

Comment:
Change the version before calling `async_update_entry` so the save it scheduled after the version is updated